File: PredatorPreyRefactor/TechnicalServices/PredictableRandomGenerator.cs
using System;

namespace TechnicalServices
{
    //This is a 'mock' implementation of IRandomGenerator, for use
    //in testing, when predictable results are needed.  You can call SetNextValues,
    //passing in as many values as needed (before the next time they are set).
    public class PredictableRandomGenerator : IRandomGenerator
    {
        private int counter = 0;
        private int[] values = null;
        public int Next(int minValue, int maxValue)
        {
            if (counter >= values.Length)
            {
                throw new Exception("Insufficient values set");
            }
            int value = values[counter];
            if (value < minValue || value >= maxValue)
            {
                throw new Exception("Value provided (" + value + ") is not within specified range");
            }
            counter++;
            return value;
        }

        public void SetNextValues(params int[] values)
        {
            counter = 0;
            this.values = values;
        }
    }
}

File: BOOM.VB/TechnicalServices/PredictableRandomGenerator.vb


Namespace TechnicalServices
	'This is a 'mock' implementation of IRandomGenerator, for use
	'in testing, when predictable results are needed.  You can call SetNextValues,
	'passing in as many values as needed (before the next time they are set).
	Public Class PredictableRandomGenerator
		Implements IRandomGenerator
		Private counter As Integer = 0
		Private values As Integer() = Nothing
        Public Function NextValue(minValue As Integer, maxValue As Integer) As Integer Implements IRandomGenerator.NextValue
            If counter >= values.Length Then
                Throw New Exception("Insufficient values set")
            End If
            Dim value As Integer = values(counter)
            If value < minValue OrElse value >= maxValue Then
                Throw New Exception("Value provided (" + value + ") is not within specified range")
            End If
            counter += 1
            Return value
        End Function

        Public Sub SetNextValues(ParamArray values As Integer())
			counter = 0
			Me.values = values
		End Sub
	End Class
End Namespace

'=======================================================
'Service provided by Telerik (www.telerik.com)
'Conversion powered by NRefactory.
'Twitter: @telerik
'Facebook: facebook.com/telerik
'=======================================================
